File: backend/src/main/java/backend/DTOs/TenantDTO.java
package backend.DTOs;

import java.time.LocalDate;

import backend.models.Lease;
import backend.models.TenantStatus;
import lombok.AllArgsConstructor;
import lombok.Data;

@AllArgsConstructor
@Data
public class TenantDTO {
    private long tenantId;
    private String firstName;
    private String lastName;
    private LocalDate dateOfBirth;
    private String email;
    private String phoneNumber;
    private String socialSecurityNumber;
    private String driversLicenseNumber;
    private String currentAddress;
    private String previousAddress;
    private String employerName;
    private String jobTitle;
    private Integer annualIncome;
    private String emergencyContactName;
    private String emergencyContactRelationship;
    private String emergencyContactPhone;
    private String referenceName;
    private String referenceRelationship;
    private String referencePhone;
    private long apartmentId;
    private Double rentDue;
    private TenantStatus tenantStatus;
    private long leaseId;
}

File: frontend/src/app/models/tenant.dto.ts
export interface TenantDTO {
  tenantId: number;
  firstName: string;
  lastName: string;
  dateOfBirth: string; // ISO date string
  email: string;
  phoneNumber: string;
  socialSecurityNumber: string;
  driversLicenseNumber: string;
  currentAddress: string;
  previousAddress?: string;
  employerName: string;
  jobTitle?: string;
  annualIncome: number;
  emergencyContactName: string;
  emergencyContactRelationship: string;
  emergencyContactPhone: string;
  referenceName: string;
  referenceRelationship: string;
  referencePhone: string;
}

